### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by Dragon-Batch** [movie-web/movie-web#903]
### What happened?

the flixhq provider does not work. the key used to decrypt the url maybe wrong. they may have changed the obfuscated script.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### Other relevant information

Vidcloud and Upcloud also do not work anymore (~2/12/24 1AM Australia). they changed the script and the regex pattern to get the key pairs does not work anymore.

**Comment by Wandang** [comment]
switch block was updated from:
```
switch (I9) {
  case 0x0:
    (II = X), (IM = t);
    break;
  case 0x1:
    (II = b), (IM = D);
    break;
  case 0x2:
    (II = x), (IM = f);
    break;
  case 0x3:
    (II = S), (IM = j);
    break;
  case 0x4:
    (II = U), (IM = G);
    break;
  case 0x5:
    (II = partKeyStartPosition_5), (IM = partKeyLength_5);
}
```
to
```
switch (R4) {
        case 0x0:
          (R5 = partKeyStartPosition_0), (R6 = partKeyLength_0);
          break;
        case 0x1:
          (R5 = partKeyStartPosition_1), (R6 = partKeyLength_1);
          break;
        case 0x2:
          (R5 = partKeyStartPosition_2), (R6 = partKeyLength_2);
          break;
        case 0x3:
          (R5 = partKeyStartPosition_3), (R6 = partKeyLength_3);
          break;
        case 0x4:
          (R5 = partKeyStartPosition_4), (R6 = partKeyLength_4);
          break;
        case 0x5:
          (R5 = partKeyStartPosition_5), (R6 = partKeyLength_5);
          break;
        case 0x6:
          (R5 = partKeyStartPosition_6), (R6 = partKeyLength_6);
          break;
        case 0x7:
          (R5 = partKeyStartPosition_7), (R6 = partKeyLength_7);
          break;
        case 0x8:
          (R5 = partKeyStartPosition_8), (R6 = partKeyLength_8);
      }
```

**Comment by Dragon-Batch** [comment]
is it possible to update the current regex to work with the new switch block?

**Comment by Dragon-Batch** [comment]
i cant seem to find where all the partKeyStartPosition_* are defined.

**Comment by Phantom-KNA** [comment]
Well this script no working anymore and not used anymore.

```
  var K = jwplayer('vidcloud-player'),
  O = [],
  v = [],
  r = Boolean(parseInt(settings.autoPlay)),
  E = $('#vidcloud-player').data('id'),
  C = $('#vidcloud-player').data('realid'),
  a = parseInt(settings.time),
  l = { channel: 'vidcloud' },
  u = true,
  T = []

  Z = (W(), '/ajax/embed-4/getSources?id=' + E),

const A = btoa(String.fromCharCode.apply(null, new Uint8Array(window[E])))

var x = CryptoJS.AES.decrypt(Z, A);
return JSON.parse(x.toString(CryptoJS.enc.Utf8))

```

**Comment by castdrian** [comment]
https://keys4.fun
https://keys4.fun/decrypt.js

**Comment by Phantom-KNA** [comment]
How obtain decrypt src:

Goto: [rabbit-player](https://rabbitstream.net/js/player/prod/e4-player.min.js?v=1707765467)
Copy all script
Goto: [Deobfuscate javascript](https://deobfuscate.relative.im/)
Paste script and click Deobfuscate, wait and save

**Comment by Dragon-Batch** [comment]
@castdrian thank you for the information.

**Comment by Dragon-Batch** [comment]
@Phantom-KNA nice time waste

**Comment by Phantom-KNA** [comment]
> @Phantom-KNA nice time waste

Maybe this can help. 
https://github.com/ghoshRitesh12/aniwatch-api/commit/c634f6e560bdb3b8dec2333439a3e285813e6532

**Comment by Dragon-Batch** [comment]
> > @Phantom-KNA nice time waste
> 
> Maybe this can help. [ghoshRitesh12/aniwatch-api@c634f6e](https://github.com/ghoshRitesh12/aniwatch-api/commit/c634f6e560bdb3b8dec2333439a3e285813e6532)

thanks

**Comment by Pasithea0** [comment]
Still not working, anyone have a quick fix?

**Comment by Phantom-KNA** [comment]
> Still not working, anyone have a quick fix?

Nothing😔

**Comment by amamadman** [comment]
> Still not working, anyone have a quick fix?

There's no point in fixing them too, the movies hosted on vidcloud/upcloud has been removed. Movies don't play on zoechip/flixhq either.

**Comment by Dragon-Batch** [comment]
yea what he said

**Comment by ownerHZH** [comment]
Is there any solution for their new encryption method?

**Comment by Dragon-Batch** [comment]
does anyone know why the download speed goes into the kilobits/s at the last ~10 segments???? it takes fucking years to download just a single video

**Comment by Phantom-KNA** [comment]
> does anyone know why the download speed goes into the kilobits/s at the last ~10 segments???? it takes fucking years to download just a single video

Contact by telegram @Phantom_KNA

**Comment by Dragon-Batch** [comment]
bro tried to get me to download ransomware on telegram 💀💀💀

**Comment by raul-gavilanes** [comment]
> bro tried to get me to download ransomware on telegram 💀💀💀

Xd brother, I'm just trying to help you, I gave you the link to download my app to watch movies. I don't know why they banned me from my previous comment, to speak privately If you were able to see my GitHub profile, I have nothing to hide, much less send you ransomware.Please @JorrinKievit You can check why it is the reason for the ban on the Phantom_KNA account

**Comment by raul-gavilanes** [comment]
![Uploading Screenshot_2024-03-13-15-41-40-565_com.google.android.gm.jpg…]()

**Comment by Dragon-Batch** [comment]
dont call me "brother". i am not your friend. nor am i related to you.

**Comment by Pasithea0** [comment]
Looks like some other sites fixed it. Still no update here?

**Comment by ownerHZH** [comment]
Still no update here?

**Comment by Ciarands** [comment]
> Still no update here?

rabbitstream likely wont be fixed for a little while.

We have a basic proof of concept and its not overly difficult to scrape, there is also a plan on how to defeat their protections at scale.
The issue is ultimately time, if we want to use this source I will have to host an API for extraction; for me, this is taking on a lot of extra responsibility for little to no reward, I don't personally use this source and its obviously all unpaid labour.
It is being worked on nonetheless, its just not a priority, I have a number of commitments elsewhere and need to keep the lights on lol

tl;dr - please be patient, we're working on it 😄

**Comment by Dragon-Batch** [comment]
does anyone know why some of the playlists are corrupt? most of the segments in them are completely missing. the either return 501 or 404 errors
